src/sql/workbench/services/notebook/sql/sqlSessionManager.ts

```diff
@@ -311,6 +325,13 @@ class SqlKernel extends Disposable implements nb.IKernel {
 		}
 		// TODO issue #2746 should ideally show a warning inside the dialog if have no data
 	}
+
+	public async disconnect(): Promise<void> {
+		if (this._path) {
+			await this._connectionManagementService.disconnect(this._path);
```

Comment:
What's path here? Can you have multiple kernels with the same path? If so disconnecting that connection might break other notebooks right?